src/Castle.Facilities.AspNet.WebApi/ISupportFacility.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright 2004-2017 Castle Project - http://www.castleproject.org/
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Castle.Facilities.AspNet.WebApi
+{
+	using Castle.MicroKernel;
+
+	public interface ISupportFacility
+	{
+		IKernel Kernel { get; }
+		bool AutoCreateLifestyleScopes { get; }
+	}
```

Comment:
I think it would be better to just pass the `IKernel` where it is needed, and pass `AutoCreateLifestyleScopes` to the controller activator on creation, you probably shouldn't let users modify this later anyway otherwise things will get screwy. At the moment this interface just makes things strange because it pretends it isn't coupling to the facility implementation.

Reply:
This field is immutable. It also saves me from passing them as separate parameters down to the activator. 

It is called interface segregation. I though this was a valid SOLID principle if you dont want to expose public members on IFacility to the activator.

Reply:
Dump it?

Reply:
I don't specifically have a problem with using an interface to expose these to the implementation of the facility even if I think the abstraction is a little pointless. If you didn't want a bunch of config booleans (if more got added), then maybe passing an IKernel and a facility configuration object would be clearer. However because of this interface you are now publicly exposing `IKernel` on the facility. I think the name of the interface is probably what bothered me the most, reading the code that uses it became a little unclear.

I should probably also warn that I'm definitely no expert in software principles or design, so this change might make perfect sense, it just seemed a little strange for passing two fields of the facility to the facility's implementation.